domino/api/src/main/java/io/quarkus/domino/ProjectDependencyResolver.java

```diff
@@ -644,9 +655,9 @@ protected Iterable<ArtifactCoords> getProjectArtifacts() {
             for (ArtifactCoords d : projectBomConstraints) {
                 final boolean collect;
                 if (includeSet.isEmpty()) {
-                    collect = d.getGroupId().startsWith(config.getProjectBom().getGroupId()) && !isExcluded(d);
+                    collect = d.getGroupId().startsWith(config.getProjectBom().getGroupId()) && !isExplicitlyExcluded(d);
                 } else {
-                    collect = !isExcluded(d) && isIncluded(d);
+                    collect = !isExplicitlyExcluded(d) && isExplicitlyIncluded(d);
```

Comment:
why do we check if the projectBomConstraint is explicitly included in this case? When discovering top level artefacts we check if they are included or not excluded.